backend/src/main/java/com/gleam/backend/game/application/GameService.java

```diff
@@ -37,9 +38,16 @@ public void addDislike(Game game) {
         gameRepository.save(game);
     }
 
-    public List<Game> getGamesByPlatform(List<String> platforms) {
+    public List<Game> getGames(String[] platforms, String prefix) {
+        var platformsList = Arrays.stream(platforms).toList();
         return gameRepository.findAll().stream()
-                .filter(game -> game.getPlatforms().stream().anyMatch(platforms::contains))
+                .filter(game -> game.getPlatforms().stream().anyMatch(platformsList::contains))
+                .filter(game -> game.getTitle().toLowerCase().startsWith(prefix.toLowerCase()))
```

Comment:
Ah, das war mir vorher gar nicht aufgefallen, aber ich würde hier nicht mit `findAll` arbeiten. Bei einer ausreichend großen Datenmenge, willst du ja nicht alle Einträge aus der Datenbank holen. Die Filterung sollte hier direkt in der Datenbank stattfinden.